app/src/main/res/layout/activity_closet.xml

```diff
@@ -1,14 +1,65 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/closet_constraint_layout"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:layout_gravity="right">
+
+    <Button
+        android:id="@+id/closet_menu_button"
+        android:layout_width="60dp"
+        android:layout_height="60dp"
+        android:layout_marginEnd="10dp"
+        android:layout_marginBottom="10dp"
+        android:background="@drawable/hamburger_button"
+        android:gravity="center_vertical|center_horizontal"
+        android:onClick="openMenu"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent" />
 
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/recycler_view"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"/>
+        android:layout_height="match_parent" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/shade_layout_holder"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <FrameLayout
+            android:id="@+id/clickable_shade"
+            android:layout_width="0dp"
+            android:layout_height="match_parent"
+            android:onClick="shadeClicked"
+            app:layout_constraintEnd_toStartOf="@+id/non_clickable_shade"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent">
+
+        </FrameLayout>
+
+        <FrameLayout
+            android:id="@+id/non_clickable_shade"
+            android:layout_width="@dimen/CLOSET_MENU_WIDTH"
+            android:layout_height="match_parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent">
+
+        </FrameLayout>
```

Comment:
Maybe I am missing something, but these two viewgroups don't appear to be used anywhere

Reply:
These are the placeholders for the shade fragments since it is simple to add the fragment into an already formatted placeholder rather than formatting it through code after adding the fragment.